src/methods/gd_lipschitz_approximation.jl

```diff
@@ -0,0 +1,198 @@
+# Date: 10/04/2024 
+# Author: Christian Varner
+# Purpose: Implement lipschitz approximation
+
+"""
+    LipschitzApproxGD{T} <: AbstractOptimizerData{T}
+
+A structure for storing data about adaptive gradient descent
+    using a Lipschitz Approximation scheme (AdGD), and the progress 
+    of its application on an optimization problem.
+
+# Fields
+
+- `name::String`, name of the solver for reference
+- `init_stepsize::T`, the initial step size for the method
+- `threshold::T`, the threshold on the norm of the gradient to induce stopping
+- `max_iterations::Int64`, the maximum allowed iterations
```

Comment:
@VChristian forgot to update the field documentation to incorporate `prev_stepsize`, `theta`, and `lipschitz_approximation`.